### PR TITLE
changed unet structure, added option for gradient dumping

### PR DESCRIPTION
**Pull request by a1302z**

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -408,11 +427,12 @@ def main(args, verbose=True, optuna_trial=None, cmd_args=None):
         model_args = {
             "encoder_name": "resnet18",
             "classes": 1,
+            "in_channels": 1,
             "activation": "sigmoid",
             "encoder_weights": None,
         }
         model = smp.Unet(**model_args)
-        model.encoder.conv1 = nn.Sequential(nn.Conv2d(1, 3, 1), model.encoder.conv1)
+        # model.encoder.conv1 = nn.Sequential(nn.Conv2d(1, 3, 1), model.encoder.conv1)
```
Comment (NiWaRe) [inline train.py:435]: Da meintest du, dass du das über in_channels im constructor machst richtig @a1302z ? +1

**Review by NiWaRe — APPROVED** [review]
Alles klar merge das dann mit den Änderungen, die ich noch gemacht hatte. @a1302z

File: train.py (hunk 1 above)
Comment (a1302z) [reply train.py:435]: genau, ist in zeile 430
